src/msw/textctrl.cpp

```diff
@@ -2720,12 +2728,50 @@ wxMenu *wxTextCtrl::MSWCreateContextMenu()
 
 void wxTextCtrl::MSWUpdateFontOnDPIChange(const wxSize& newDPI)
 {
-    // Don't do anything for the rich edit controls, they (somehow?) update
-    // their appearance on their own and changing their HFONT, as the base
-    // class version does, would reset all the styles used by them when the DPI
-    // changes, which is unwanted.
+    // Don't use MSWUpdateFontOnDPIChange for the rich edit controls, they
+    // (somehow?) update their appearance on their own and changing their
+    // HFONT, as the base class version does, would reset all the styles used
+    // by them when the DPI changes, which is unwanted.
     if ( !IsRich() )
+    {
         wxTextCtrlBase::MSWUpdateFontOnDPIChange(newDPI);
+    }
+    // If the rich control is created on a screen with non-system DPI, an
+    // initial scaling is applied. This needs to be reset after the first DPI
+    // change. Reapply the styles to the text.
+    else if ( m_richBaseDPI != m_richUsedDPI )
+    {
+        wxEventBlocker block(this, wxEVT_TEXT);
+
+        wxSize baseDPI = m_richBaseDPI;
```

Comment:
```suggestion
        const wxSize baseDPI = m_richBaseDPI;
```
Just to make it more clear what's going on here.